client/components/site-selector/index.jsx

```diff
@@ -472,6 +474,10 @@ const navigateToSite =
 			const site = getSite( state, siteId );
 			debug( 'getPathnameForSite', siteId, site );
 
+			if ( isJetpackCloud() && ! site && showAgencyDashboard( state ) ) {
+				return page.redirect( '/dashboard' );
+			}
+
```

Comment:
This component is generic enough to be used across different apps (e.g., WordPress.com and Jetpack cloud), so it is not optimal to handle this particular case for Jetpack cloud.

Wouldn't it be possible to move this logic up the call stack? I suspect that we might be able to find a better place in some component that is specific to Jetpack cloud.

Reply:
I don't think we can find a better place than this since this redirection happens on click of Switch Site and we have the logic currently in the navigate function which gets called on click. This function also handles redirection to pages like `/backup`, `/scan` etc. Basically, we have to redirect the user to `/dashboard` before it gets redirected to other pages.

Reply:
Understood. I checked that these changes are not affecting WordPress.com's behavior in any way.